.github/workflows/ci.yml

```diff
@@ -13,40 +13,38 @@ jobs:
     strategy:
       matrix:
         php-version:
-         - '8.2'
-         - '8.3'
-         - '8.4'
-        dependencies: ['']
+          - '8.2'
+          - '8.3'
+          - '8.4'
+        dependencies:
+          - highest
         include:
-          - { php-version: '8.2', dependencies: '--prefer-lowest' }
+          - php-version: '8.2'
+            dependencies: lowest
 
     name: PHP ${{ matrix.php-version }} ${{ matrix.dependencies }}
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php-version }}
-          extensions:
           coverage: pcov
-          tools: composer:v2
+          tools: composer:v2,composer-normalize
 
-      - name: Cache dependencies
-        uses: actions/cache@v3
-        with:
-          path: ~/.cache/composer
-          key: composer-${{ matrix.php-version }}-${{ hashFiles('**/composer.*') }}-${{ matrix.dependencies }}
-          restore-keys: |
-            composer-${{ matrix.php-version }}-${{ hashFiles('**/composer.*') }}-
-            composer-${{ matrix.php-version }}-
-            composer-
+      - name: Validate composer.json
+        run: |
+          composer validate --strict
+          composer-normalize --diff --dry-run
```

Comment:
Thanks for the PR! I have to say I'm not super pumped about the global `composer-normalize`. It feels like a good idea, but I tried the same with php-cs-fixer, and found that for less often updated packages it creates a annoying version gap, when a version installed via `composer.json` is newer, so CI build may not reliably fail on errors.